component/otelcol/receiver/kafka/kafka.go

```diff
@@ -88,20 +94,28 @@ func (args *Arguments) SetToDefault() {
 
 // Convert implements receiver.Arguments.
 func (args Arguments) Convert() (otelcomponent.Config, error) {
-	return &kafkareceiver.Config{
-		Brokers:         args.Brokers,
-		ProtocolVersion: args.ProtocolVersion,
-		Topic:           args.Topic,
-		Encoding:        args.Encoding,
-		GroupID:         args.GroupID,
-		ClientID:        args.ClientID,
-		InitialOffset:   args.InitialOffset,
-
-		Authentication: args.Authentication.Convert(),
-		Metadata:       args.Metadata.Convert(),
-		AutoCommit:     args.AutoCommit.Convert(),
-		MessageMarking: args.MessageMarking.Convert(),
-	}, nil
+	input := make(map[string]interface{})
+	input["auth"] = args.Authentication.Convert()
+
+	var result kafkareceiver.Config
+	err := mapstructure.Decode(input, &result)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Just curious, is this because the kafka auth struct got moved to the `internal` package, so we can't instantiate it directly anymore and now need to make the map and convert it indirectly?

Reply:
Yes, that's correct.